### DISCUSSION TITLE
Contributor call: 13 May 2023

### DISCUSSION THREAD

**Discussion by l3xander** [freesewing/freesewing#3934] [Support] [answered]
You are all invited to the FreeSewing Contributor Call.
Please leave a comment here or [on Discord](https://discord.freesewing.org/) to add things to the agenda.
This issue will be updated after the call with notes and to-dos.

### About
FreeSewing's contributor call is an informal hangout/chat/meeting that is open to all FreeSewing contributors and users alike.
The call is organized/hosted by @tangerineshark.
The call does not have a fixed structure or agenda items. Everyone is free to suggest agenda items, ask questions, or bring ideas.

**Details**
Location: https://discord.freesewing.org (meeting room: 🔊`voice chat`)
Date: May 13 2023 4pm UTC, which can be found in your local time zone [here](https://savvytime.com/converter/utc-to-belgium-antwerpen-ca-los-angeles-il-chicago-ny-new-york-city-brazil-sao-paulo-united-kingdom-london-france-paris-south-africa-johannesburg-msd-india-mumbai-hkt-japan-tokyo-australia-sydney/may-13-2023/4pm).
Or check the Events tab in Discord!
The meetings will always be at 6pm, Antwerp time.

### Agenda
- Roadmap, issues, and shortening the v3 critical path
- Tone and language in documentation, #3842 and #3750
- Our Discord server was boosted, and that's kind of a big deal
- proposal: Allow plugins to export options that can then be controlled at runtime (eg: annotation plugin might have an options to hide scalebox/logo/whatever)
- v3 Curated collections 
- macronames should always be lowercase
- no more *gist* in v3. Instead we'll have settings and ui state.
- Should we provide React component(s) to render patterns?
- Granularity in the `details` setting

**Comment by BenJamesBen** [comment]
The tone we want in sewing related documentation and personal language in documentation.
https://github.com/freesewing/freesewing/pull/3842
https://github.com/freesewing/freesewing/issues/3750

**Comment by BenJamesBen** [comment]
(There are 3 existing macros with names containing a non-lowercase character: `bartackAlong`, `bartackFractionAlong`, and `sewTogether`.)

**Reply by joostdecock** [reply]
Crap, I forgot about those. Stupid Joost and his stupid long macro names.
Alright, then let's just make them case-insensitive, because my main concern is that people shouldn't have to remember the casing. That's why I was thinking all lowercase, but something like `bartackalong` would then conflict with `Point.shiftAlong` so that's not great either.

**Comment by tangerineshark** [comment, marked as answer]
### Notes

Roadmap, issues, and shortening the v3 critical path

- @joostdecock: Goal: get v3 done ASAP, so people can use it. So, prioritizing things based on whether or not they are critical to rolling out v3. If they are, do it ASAP. If they're good but not critical, put it on the roadmap. And if we don't want to do something (or don't want to do it now), close the issue. (@tangerineshark editorializing here for folks worried about closing unresolved issues: if it's actually something we will want later, it'll probably raise itself again.) 
- @BenJamesBen: how do we tell which items are on the v3 critical path? Answer from Joost: if something's a breaking change, it's on the v3 critical path; if it's not, it might be something we move to later.
- goal is to get the core stable

Tone and language in documentation, https://github.com/freesewing/freesewing/pull/3842 and https://github.com/freesewing/freesewing/issues/3750

- Based on thoughts from last call, we're going to try to show who wrote the docs more, by noting it on the page, in the front matter, etc., because we like the personal quality of individual documentation written by different people, but it can be confusing if you expect all docs to have the same voice
- general principles: don't edit documentation for tone, unless it causes genuine confusion
- do edit for actual problems (e.g., incorrect instructions), but if it changes the tone dramatically, it wouldn't hurt to check with the author where feasible

Our Discord server was boosted, and that's kind of a big deal

- @lexander2002 notes that two members have boosted our server, giving us access to cooler features, so shout-out to H0rt3ns14 and usablehoney - you're awesome! <3

proposal: Allow plugins to export options that can then be controlled at runtime (eg: annotation plugin might have an options to hide scalebox/logo/whatever)

v3 Curated collections

- What's a curated collection? It's a curated set of measurements from an actual person who has opted to share their measies publicly (there's currently a setting where you can share your measies with FS, but we would probably have another opt-in to share these curated collections more publicly)
- there might be UI impacts depending on whether you use your own measies or a curated collection (tagging @joostdecock to add additional details here as needed)

macronames should always be lowercase

- macronames will become case insensitive: right now, some are lowercase and some are CamelCase; this will solve for that so that both work

no more gist in v3. Instead we'll have settings and ui state.

Should we provide React component(s) to render patterns?

- @eriese says yes

Granularity in the details setting

(okay, @tangerineshark's notes got kinda bad here, so feel free to ping me with additional details if you want them added. Failures in explanation and misrepresentations are entirely my own.)

**Comment by joostdecock** [comment]
Closing to increase visibility of active discussions
